src/transformers/modeling_utils.py

```diff
@@ -1649,6 +1652,13 @@ def _check_and_enable_sdpa(cls, config, hard_check_only: bool = False) -> Pretra
 
         if not hard_check_only:
             config._attn_implementation = "sdpa"
+
+        if torch.version.hip is not None and config._attn_implementation == "sdpa" and device_map == "auto":
```

Comment:
Other device_map may fail as well no? e.g. if manually splitting the layers on several GPUs

Reply:
Yes, it could happen on other types of device map where it could use multiple devices. Updated to check the `torch.cuda.device_count() > 1`.